src/Http/Routing/src/Constraints/FileNameRouteConstraint.cs

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Globalization;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.Routing.Constraints
+{
+    /// <summary>
+    /// Constrains a route parameter to represent only filename values.
+    /// </summary>
+    /// <remarks>
+    /// <para>
+    /// This constraint can be used with a low priority route to disambiguate for requests for
+    /// static files versus dynamic content served from the application.
+    /// </para>
+    /// <para>
+    /// This constraint determines whether a route value represents a file name by examining
+    /// the last URL Path segment of the value (delimited by <c>/</c>). The last segment
+    /// must contain the <c>.</c> character surrounded by characters on both sides. If the segment
+    /// contains multiple <c>.</c>characters, then the last one will be considered.
+    /// </para>
+    /// <para>
+    /// If the route value does not contain a <c>/</c> then the entire value will be interpreted
+    /// as the last segment.
+    /// </para>
+    /// <para>
+    /// The <see cref="FileNameRouteConstraint"/> does not attempt to validate that the value contains
+    /// a legal file name for the current operating system.
+    /// </para>
+    /// <para>
+    /// <list type="bullet">  
```

Comment:
I'm interested to know how all this content surfaces to users. It gets injected into the auto-generated API documentation web pages, does it?

Reply:
That's the idea at least. I don't have an example handy.